sdk/storage/azure-storage-blob/azure/storage/blob/blob_service_client.py

```diff
@@ -0,0 +1,469 @@
+            hour_metrics=hour_metrics,
+            minute_metrics=minute_metrics,
+            cors=cors,
+            default_service_version=target_version,
+            delete_retention_policy=delete_retention_policy,
+            static_website=static_website
+        )
+        try:
+            self._client.service.set_properties(props, timeout=timeout, **kwargs)
+        except StorageErrorException as error:
+            process_storage_error(error)
+
+    def list_containers(
+            self, name_starts_with=None,  # type: Optional[str]
+            include_metadata=False,  # type: Optional[bool]
+            marker=None,  # type: Optional[str]
+            results_per_page=None,  # type: Optional[int]
+            timeout=None,  # type: Optional[int]
+            **kwargs
+        ):
+        # type: (...) -> ContainerPropertiesPaged
+        """Returns a generator to list the containers under the specified account.
+        The generator will lazily follow the continuation tokens returned by
+        the service and stop when all containers have been returned.
+
+        :param str name_starts_with:
+            Filters the results to return only containers whose names
+            begin with the specified prefix.
+        :param bool include_metadata:
+            Specifies that container metadata be returned in the response.
+        :param str marker:
+            An opaque continuation token. This value can be retrieved from the
+            next_marker field of a previous generator object. If specified,
+            this generator will begin returning results from this point.
+        :param int results_per_page:
+            The maximum number of container names to retrieve per API
+            call. If the request does not specify the server will return up to 5,000 items.
+        :param int timeout:
+            The timeout parameter is expressed in seconds.
+        :returns: An iterable (auto-paging) of ContainerProperties.
+        :rtype: ~azure.core.blob.models.ContainerPropertiesPaged
+        """
+        include = 'metadata' if include_metadata else None
+        command = functools.partial(
+            self._client.service.list_containers_segment,
+            prefix=name_starts_with,
+            include=include,
+            timeout=timeout,
+            **kwargs)
+        return ContainerPropertiesPaged(
+            command, prefix=name_starts_with, results_per_page=results_per_page, marker=marker)
+
+    def create_container(
+            self, name,  # type: str
+            metadata=None,  # type: Optional[Dict[str, str]]
+            public_access=None,  # type: Optional[Union[PublicAccess, str]]
+            timeout=None,  # type: Optional[int]
+            **kwargs
+        ):
+        # type: (...) -> ContainerClient
+        """Creates a new container under the specified account. If the container
+        with the same name already exists, the operation fails. Returns a client with
```

Comment:
clarify whether a client is still returned if the operation fails.

Reply:
All operation failures in the new SDK raise an exception (i.e. we don't return `None` or `False`)